lib/global-styles.php

```diff
@@ -137,11 +137,23 @@ function_exists( 'gutenberg_is_edit_site_page' ) &&
 	// Copied from get_block_editor_settings() at wordpress-develop/block-editor.php.
 	$settings['__experimentalFeatures'] = $consolidated->get_settings();
 	if ( isset( $settings['__experimentalFeatures']['color']['palette'] ) ) {
-		$settings['colors'] = $settings['__experimentalFeatures']['color']['palette'];
+		$colors_by_origin   = $settings['__experimentalFeatures']['color']['palette'];
+		$settings['colors'] = isset( $colors_by_origin['user'] ) ?
```

Comment:
This is a behaviour change: the visible color palette becomes either the user colors, the theme, or core. It'd be good to clarify desired behaviour but my understanding is that we want it to work this way:

- either show theme or core colors in the UI
- user colors are added _in addition_ to the previous ones.

Reply:
Related https://github.com/WordPress/gutenberg/issues/32027

Reply:
> This is a behavior change: the visible color palette becomes either the user colors, the theme's, or core's.

I don't think there is any behavior change. Currently, our merge logic already works like this user colors replace theme or core colors and are not added in addition. But when the user is changing colors the user starts from theme colors so if the user adds a color the theme colors are also part of the user colors. We also don't allow the user to delete theme colors. All these behaviors are kept so I think there is no change.

Reply:
Oh, because the way user colors work in the site editor: they copy the existing colors array. Right, thanks for confirming.